core/src/main/scala/pickling/AllPicklers.scala

```diff
@@ -7,9 +7,27 @@ object AllPicklers extends CorePicklersUnpicklers
 
 object all extends CorePicklersUnpicklers {
 
-  implicit class PickleOps[T](picklee: T) {
-    def pickle(implicit format: PickleFormat): format.PickleType = macro Compat.PickleMacros_pickle[T]
-    def pickleInto(builder: PBuilder): Unit = macro Compat.PickleMacros_pickleInto[T]
+  implicit class PickleOps[T: SPickler](picklee: T) {
+    def pickle(implicit format: PickleFormat): format.PickleType = {
```

Comment:
could we also put a `def pickle[T](picklee: T)(implicit pickler: SPickler[T], format: PickleFormat): format.PickleType` into the pickling package object, so you can do `scala.pickling.pickle(t)` without having to use the implicit? Sometimes it's nice to be explicit. Also, you probably conflict with #267 here, I think @eed3si9n changed PickleOps around a bit.